components/prophet/election/election.go

```diff
@@ -29,7 +29,7 @@ var (
 // Elector a leader elector
 type Elector interface {
 	// CreateLeadship create a leadership
-	CreateLeadship(purpose string, nodeName, nodeValue string, allowCampaign bool, becomeLeader, becomeFollower func(string) bool) *Leadership
+	CreateLeadship(purpose string, nodeName, nodeValue string, isProphet bool, becomeLeaderFunc, becomeFollowerFunc func(string) bool) *Leadership
```

Comment:
Elector is a relatively independent module that can be used in isolation from prophet. Suggest that allowCampaign be renamed allowBecomeLeader